### PR TITLE
Add docs for BNNS lib integration

### PR DESCRIPTION
**Pull request by dlexplorer**
BNNS integration docs

### CODE REVIEW

File: docs/deploy/bnns.rst [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+==========================
+**Author**: `Egor Churaev <https://github.com/echuraev>`_
+
+Introduction
+------------
+
+Apple BNNS library is a collection of functions that can be used to construct neural networks
+for inference (and train). It’s supported in macOS, iOS, tvOS, and watchOS. BNNS provides
+primitives executed on all CPU supported on those platforms and optimized for high performance
+and low-energy consumption. This integration will offload as many operators as possible from Relay to BNNS.
```
Comment (apeskov) [inline docs/deploy/bnns.rst:28]: Also should be noted:
* BNNS is a part of platform API and available on all modern apple systems. Does't required any external dependencies.

Possible benefits:
* Allow to use Apple private HW optimisation which doesn't accessible outside. Like AMX apple cpu extension.  
* Reducing binary size. No need to keep TVM compiled kernels, just transfer it into system BNNS call.   


> This integration will offload as many operators as possible
It's better to formulate "This integration allow to transfer operation execution from Relay to BNNS." 
Greedy BNNS notation is an option, not mandatory pass.

File: docs/deploy/bnns.rst [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+
+This guide will demonstrate how to build TVM with BNNS BYOC and runtime enabled. It will also provide example
+code to compile and run models using BNNS runtime. Finally, we document the supported operators.
```
Comment (apeskov) [inline docs/deploy/bnns.rst:31]: About term **BNNS BYOC**:
I'm not sure that "Bring Your Own Code" (BYOC) is an official name of component lice "code generator" or "runtime". From my point of view that is a technic and it cannot be built.

I suggest to use terms BNNS Codegen and Runtime modules.

File: docs/deploy/bnns.rst [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+
+Building TVM with BNNS support
+----------------------------------
+
+To turn on TVM BNNS BYOC and TVM BNNS runtime you need to turn on the only USE_BNNS flag
```
Comment (apeskov) [inline docs/deploy/bnns.rst:36]: Also:
It will search the default Accelerate frameworks on your current target SDK. The minimal versions of required SDK is macOS 11.0, iOS 14.0, tvOS 14.0 and watchOS 7.0.

File: docs/deploy/bnns.rst [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+
+Operator support
+----------------
+
++------------------------+------------------------------------------------------------------------------+
+|       Relay Node       |              Remarks                                                         |
++========================+==============================================================================+
+| nn.conv2d              |                                                                              |
++------------------------+------------------------------------------------------------------------------+
+| nn.batch_norm          |                                                                              |
```
Comment (apeskov) [inline docs/deploy/bnns.rst:159]: batch norm is not supported. We expect it will be merged into nearest conv/dense primitive.

File: docs/deploy/bnns.rst [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+    set(USE_BNNS ON)
+
+BNNS partitioning of Relay graph
+----------------------------------------
+
+Operations to be offloaded on BNNS execution must be annotated before passing of module for compilation.
+All opps annotated by `partition_for_bnns` will be offloaded for BNNS execution. The rest of the ops
+will go through the LLVM compilation and code generation.
+
+Important note: `partition_for_bnns` requires to freeze parameters before the annotations of ops for BNNS.
```
Comment (apeskov) [inline docs/deploy/bnns.rst:54]: Have to reveal the reason of this requirements more deeper.
BNNS support primitives only with constant weights. To meet this requirements we have to map constants to related tensor abstraction in relay representation. For this purposes you may need to call ONNX importer with special flag  "freeze_params=True" or performer binding manually. In general cases all relay importers don't do that be default. For your convenience we perform that inside  "partition_for_bnns" helper if will provide params dictionary as argument.

File: docs/deploy/bnns.rst [hunk 6]
```diff
@@ -0,0 +1,172 @@
...
++========================+==============================================================================+
+| nn.conv2d              |                                                                              |
++------------------------+------------------------------------------------------------------------------+
+| nn.batch_norm          |                                                                              |
++------------------------+------------------------------------------------------------------------------+
+| nn.dense               |                                                                              |
++------------------------+------------------------------------------------------------------------------+
+| nn.batch_matmul        |                                                                              |
++------------------------+------------------------------------------------------------------------------+
+| nn.bias_add            | Supported by BNNS integration only as a part of nn.conv2d or nn.dense fusion |
```
Comment (apeskov) [inline docs/deploy/bnns.rst:165]: > ...as a part of nn.conv2d or nn.dense fusion
Better: ...as a **bias** part of nn.conv2d or nn.dense fusion

We allow to fuse only if it is a real bias. Broadcasted tensor like [1, OC, 1, 1]

File: docs/deploy/bnns.rst (hunk 1 above)
Comment (dlexplorer) [reply docs/deploy/bnns.rst:28]: Addressed. slightly modified - removed mentioning of binary size. It is questionable statement. TVM code gen will produce much less size than it will be in BNS runtime.

File: docs/deploy/bnns.rst (hunk 2 above)
Comment (dlexplorer) [reply docs/deploy/bnns.rst:31]: DONE

File: docs/deploy/bnns.rst (hunk 3 above)
Comment (dlexplorer) [reply docs/deploy/bnns.rst:36]: DONE

File: docs/deploy/bnns.rst (hunk 4 above)
Comment (dlexplorer) [reply docs/deploy/bnns.rst:159]: mentioned that batch_norm is supported only as part of conv2d-batch_norm pattern

File: docs/deploy/bnns.rst (hunk 5 above)
Comment (dlexplorer) [reply docs/deploy/bnns.rst:54]: rephrased, please take a look

File: docs/deploy/bnns.rst (hunk 6 above)
Comment (dlexplorer) [reply docs/deploy/bnns.rst:165]: do not feel a difference but changed

File: docs/deploy/bnns.rst [hunk 7]
```diff
@@ -51,9 +60,11 @@ Operations to be offloaded on BNNS execution must be annotated before passing of
 All opps annotated by `partition_for_bnns` will be offloaded for BNNS execution. The rest of the ops
 will go through the LLVM compilation and code generation.
 
-Important note: `partition_for_bnns` requires to freeze parameters before the annotations of ops for BNNS.
-This can be done by passing of special parameters to some importers like `from_onnx(onnx_model, freeze_params=True)`
-or to pass params to `partition_for_bnns` function
+Important note: BNNS support primitives only with constant weights. To satisfy this requirements we have
+to map constants to related tensor abstraction in relay representation. To freeze tensors and operate
+with them as with constants you may need to call ONNX importer with special flag "freeze_params=True"
+or performer binding manually. In general cases all relay importers don't do that be default.
```
Comment (apeskov) [inline docs/deploy/bnns.rst:66]:
```suggestion
or performer binding manually. In general cases all relay importers don't do that by default.
```

File: docs/deploy/bnns.rst [hunk 8]
```diff
@@ -156,13 +167,14 @@ Operator support
...
 +------------------------+------------------------------------------------------------------------------+
 | nn.dense               |                                                                              |
 +------------------------+------------------------------------------------------------------------------+
 | nn.batch_matmul        |                                                                              |
 +------------------------+------------------------------------------------------------------------------+
-| nn.bias_add            | Supported by BNNS integration only as a part of nn.conv2d or nn.dense fusion |
+| nn.bias_add            | Supported by BNNS integration only as a bias part of nn.conv2d or nn.dense   |
+|                        | fusion                                                                       |
 +------------------------+------------------------------------------------------------------------------+
 | add                    | Supported by BNNS integration only as a part of nn.conv2d or nn.dense fusion |
```
Comment (apeskov) [inline docs/deploy/bnns.rst:179]:
```suggestion
| add                    | Supported by BNNS integration only as a bias part of nn.conv2d or nn.dense fusion |
```

File: docs/deploy/bnns.rst (hunk 4 above)
Comment (apeskov) [reply docs/deploy/bnns.rst]: I still recommend to remove this line with BN
